src/rust/ide/src/ide/integration.rs

```diff
@@ -1133,6 +1147,24 @@ impl Model {
             }
         });
     }
+
+    fn visualization_preprocessor_changed(&self, node_id:graph_editor::NodeId, code:&graph_editor::data::EnsoCode)
+    -> FallibleResult {
+        if let Some(visualization) = self.visualizations.get_copied(&node_id) {
+            let logger      = self.logger.clone_ref();
+            let controller  = self.graph.clone_ref();
+            let code_string = AsRef::<String>::as_ref(code).clone();
```

Comment:
what does this `clone` do? Doesn't it just copy a reference?

Reply:
No. Calling clone on a reference will clone the content, otherwise such a code wouldn't work as expected
```rust
fn foo(x:&SomeBigStruct) -> SomeBigStruct {
    let c = x.clone();
   ...
}
```
And I checked it: if I remove clone, I get an error saying "expected String, got reference".

Reply:
Oh, got it. Can we then simplify the code and add a method `to_string` to `enso::Code` please?

Reply:
the `enso::Code` is generated by `im_string_newtype` - I make that macro producing `Display`, and with Display we get also to_string.